packages/fxa-auth-db-mysql/db-server/test/backend/db_tests.js

```diff
@@ -1474,6 +1474,102 @@ module.exports = function (config, DB) {
+
+        uid = accountData.uid
+        anotherUid = anotherAccountData.uid
+
+        return P.all([
+          db.createSessionToken(session1.tokenId, session1),
+          db.createSessionToken(session2.tokenId, session2),
+          db.createSessionToken(session3.tokenId, session3),
+          db.createSessionToken(anotherAccountSession.tokenId, anotherAccountSession)
+        ])
+        // Don't parallelize these, the order of them matters
+        // because they record timestamps in the db.
+          .then(() => insert(uid, addr, evA, session2.tokenId).delay(10))
+          .then(() => insert(uid, addr, evB, session1.tokenId).delay(10))
+          .then(() => insert(uid, addr, evC).delay(10))
+          .then(() => insert(anotherUid, addr, evA, anotherAccountSession.tokenId).delay(10))
+      })
+
+      it('should get security event', () => {
+        return db.securityEventsByUid(uid)
+          .then( results => {
+            assert.lengthOf(results, 3)
+            // The most recent event is returned first.
+            assert.equal(results[0].name, evC, 'correct event name')
+            assert.equal(!! results[0].verified, true, 'event without a session is already verified')
+            assert.isBelow(results[0].createdAt, Date.now())
+            assert.equal(results[1].name, evB, 'correct event name')
+            assert.equal(!! results[1].verified, false, 'second session is not verified yet')
+            assert.isBelow(results[1].createdAt, results[0].createdAt)
+            assert.equal(results[2].name, evA, 'correct event name')
+            assert.equal(!! results[2].verified, true, 'first session is already verified')
+            assert.isBelow(results[2].createdAt, results[1].createdAt)
+          })
+      })
+
+      it('should get security event for another account', () => {
+        return db.securityEventsByUid(anotherUid)
+          .then( results => {
+            assert.lengthOf(results, 1)
+            assert.equal(results[0].name, evA, 'correct event name')
+            assert.equal(results[0].verified, false, 'this session is not verified')
+            assert.isBelow(results[0].createdAt, Date.now())
+          })
+      })
+
+      it('should get event after session verified', () => {
+        return db.verifyTokens(session1.tokenVerificationId, { uid })
+          .then(() => db.securityEventsByUid(uid))
+          .then( results => {
+            assert.lengthOf(results, 3)
+            assert.isTrue(!! results[0].verified)
+            assert.isTrue(!! results[1].verified)
+            assert.isTrue(!! results[2].verified)
+          })
+      })
+
+      it('should give no securityEvent with new unknown uid', () => {
+        const unknownUid = newUuid()
+        return db.securityEventsByUid(unknownUid)
+          .then( results => assert.lengthOf(results, 0))
+      })
+
+      it('should delete events when account is deleted', () => {
```

Comment:
We should be testing `should delete events`. I would modify this to call `db.deleteSecurityEventsByUid`. It looks like that implementation is missing for the memory database.